src/content/1.7/release/release-a-native-PS-module.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Release a native PrestaShop module
+weight: 3
+---
+
+# Release a new version of a native PrestaShop module
+
+Some modules are embedded within PrestaShop ZIP archives and loaded through Composer package manager. They consequently are listed in the `composer.json` of the Core project.
+
+This is what we call "native PrestaShop modules". The source code of these modules is hosted on GitHub.
+
+This page explains what is needed to release these modules.
+
+## Requirements
+
+### Mandatory requirements
+
+The module should be valid which means:
+
+- it works as expected
+- it complies with modules [contribution guidelines][contribution-guidelines]: license headers must be valid and necessary files (LICENSE, CONTRIBUTORS) must be embedded
+
+### Recommended requirements
+
+It is recommended:
+
+- to have enabled [Release Drafter](https://github.com/release-drafter/release-drafter) on the module to automatize the process of creating a new GitHub release
+- have enabled some Continous Integration tools such as phpstan or php-cs-fixer on the module
+
+These tools will help releasing a clean module in a smooth manner but the release can happen without them.
+
+## Release process
+
+
+When multiple improvements (bug fixes, enhancements or new features) have been merged into the `dev` branch of a module, we can merge them into `master` to deliver a new release of the module.
+
+### Choosing the new version number
+
+We follow [SemVer](https://semver.org/) guidelines:
+
+If the new version of the module is backward compatible, you can increase the version number either by a minor version or by a patch version.
+
+However, if the new version of the module breaks backward compatibility, it must be a new major version.
+
+### Update version number in the code
+
+Create a Pull Request targeting `dev` branch. In this PR:
+
+Update the module version in the following files:
+
+- `config.xml`
+- `<module_name>.php`
+
+Commit the changes with git message:
+```
+// Version updated to v<new_version>
+```
+Get this PR approved and merge it into `dev`. See an [example](https://github.com/PrestaShop/ps_shoppingcart/pull/50/) of such PR. 
+
+### Merge dev into master
+
+Then create a new PR to merge branch `dev` into branch `master`. See an [example](https://github.com/PrestaShop/ps_shoppingcart/pull/51) of such PR.
+
+Kindly ask the QA team to perform a release test on this PR, warning them that this PR is a release PR and consequently they need to check the whole module behavior.
+
+If QA validates the PR, it can be merged, which will update the `master` branch with the new code changes from `dev`.
+
+### Create a GitHub Release
+
+If you have enabled Release Drafter, it will create a draft release for you. You can review it and publish it.
+
+Else, you need to do it manually:
+
+Create a new tag with "v<new_version>" that targets the last commit on `master` and push it. This will publish the module version on https://packagist.org/ .
+
```

Comment:
```suggestion
```

Done with `Create a release`

Reply:
No, it's 2 different things. Creating the git tag is an action inside git metadata while creating a release is creating a new item in GitHub database.

Maybe I can use better words to make this unambiguous ?

Reply:
Fake news ;) 

When you create a release on Github, that create a tag.

![image](https://user-images.githubusercontent.com/1533248/85274680-914fe500-b47f-11ea-8070-225c3cb61904.png)

Reply:
Ah, this is what you meant 😄 but this is still 2 different things. However GitHub is able to "kill 2 birds with 1 stone".

But I think it's better to explain the 2 things in 2 separate lines, just like here https://github.com/PrestaShop/docs/pull/593

This is a documentation, it aims to be as complete as possible 😉 it's better IMO to explain each thing separately so people can grasp what each step is about.

However I can update the PR to provide a tip "by the way, GitHub can do this for you with the option ..."

Reply:
Ok, for the last option ;)

Reply:
Fixed in https://github.com/PrestaShop/docs/pull/591/commits/415b4cd1258516cc371bb750269e85a68546e04b